### ISSUE TITLE
Can't compile on the latest nightly

### ISSUE THREAD

**Issue by RoxasShadow** [benashford/rs-es#80]
`rustc 1.11.0-nightly (1ab87b65a 2016-07-02)`

```
   Compiling rs-es v0.4.4
/Users/giovanni/.cargo/registry/src/github.com-1ecc6299db9ec823/rs-es-0.4.4/src/operations/search/aggregations/bucket.rs:106:17: 106:26 error: overflow evaluating the requirement `query::Query: serde::Serialize` [E0275]
/Users/giovanni/.cargo/registry/src/github.com-1ecc6299db9ec823/rs-es-0.4.4/src/operations/search/aggregations/bucket.rs:106 #[derive(Debug, Serialize)]
                                                                                                                                             ^~~~~~~~~
/Users/giovanni/.cargo/registry/src/github.com-1ecc6299db9ec823/rs-es-0.4.4/src/operations/search/aggregations/bucket.rs:106:17: 106:26 help: run `rustc --explain E0275` to see a detailed explanation
/Users/giovanni/.cargo/registry/src/github.com-1ecc6299db9ec823/rs-es-0.4.4/src/operations/search/aggregations/bucket.rs:106:17: 106:26 note: required by `serde::Serialize`
error: aborting due to previous error
error: Could not compile `rs-es`.
```

P.S.: New versions of serde and its dependencies have been released.

**Comment by benashford** [comment]
Thanks for alerting me to this.  I should really setup a trigger that runs the Travis job on a nightly basis to test for these things.

The bad news, this doesn't just affect nightly, this also affects stable and beta Rust releases too.  Occasional build breakages on nightly are part of the risk of using nightly, but stable shouldn't start breaking without warning.  It looks like (for stable) it is a bug introduced in a patch release of one or more of the Serde projects.  Unfortunately this isn't the first time this has happened.  I think from now on I'll have to release `rs-es` against specific pinned versions of the Serde libraries, and not let patch releases be automatically used.  Hopefully this should keep stable stable, but more manual bumping will be needed to keep nightly building.

I'll make the stable builds stable first, then come back to the nightly build in due course.

**Comment by benashford** [comment]
v0.4.5 restores stable and beta.

I'll look into restoring nightly next.

**Comment by RoxasShadow** [comment]
Thanks a lot.

**Comment by RoxasShadow** [comment]
Just tried compiling with the latest nightly. The bad is that it still does not compile. The good thing is that the error is the same, so probably nothing has been broken additionally.

**Comment by benashford** [comment]
Yes, I've had to pin serde_codegen and related projects at 0.7.11 because 0.7.12 has seemingly introduced a bug that prevents rs-es from compiling.  And unfortunately 0.7.11 is out-of-date with the latest Rust nightlies, but it does still build on Stable on Beta channels.

I haven't yet fully pinned it down, and once I have I'll open an Issue on the Serde project, but I'll confirm it first in case I'm wrong.  I've just tried 0.7.13 of Serde, and unfortunately that still has the same problem.

The build error can be seen here: https://travis-ci.org/benashford/rs-es/builds/143149102

I haven't been able to put much time into this, but I'll hopefully get a bit of time in the next few days.

**Comment by benashford** [comment]
The Serde project acknowledge this was a breaking change introduced in a patch release: https://github.com/serde-rs/serde/issues/441 I'm now looking into applying the appropriate fix.

I need to either: a) manually specify the trait bounds, or b) opt-out of deriving Serialize on certain key types and do it manually instead.

**Comment by benashford** [comment]
Fixed in `0.4.6`

**Comment by RoxasShadow** [comment]
Great job!
